test/boost/secondary_index_test.cc

```diff
@@ -1529,26 +1550,301 @@ SEASTAR_TEST_CASE(test_token_order) {
+            msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) > {} AND token(pk) < {} ALLOW FILTERING", testset_tokens[1], testset_tokens[4]).c_str());
+            assert_that(msg).is_rows().with_rows(get_result_rows(2, 3));
+
+            // inclusive-infinity range
+            msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) >= {} ALLOW FILTERING", testset_tokens[3]).c_str());
+            assert_that(msg).is_rows().with_rows(get_result_rows(3, testset_pks.size() - 1));
+
+            // exclusive-infinity range
+            msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) > {} ALLOW FILTERING", testset_tokens[3]).c_str());
+            assert_that(msg).is_rows().with_rows(get_result_rows(4, testset_pks.size() - 1));
+
+            // infinity-inclusive range
+            msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) <= {} ALLOW FILTERING", testset_tokens[3]).c_str());
+            assert_that(msg).is_rows().with_rows(get_result_rows(0, 3));
+
+            // infinity-exclusive range
+            msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) < {} ALLOW FILTERING", testset_tokens[3]).c_str());
+            assert_that(msg).is_rows().with_rows(get_result_rows(0, 2));
+
+            // negative-positive range
+            auto start_index =  3;
+            auto end_index = 5;
+            assert(testset_tokens[start_index] < 0);
+            assert(testset_tokens[end_index] > 0);
+
+            msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) >= {} AND token(pk) <= {} ALLOW FILTERING", testset_tokens[start_index], testset_tokens[end_index]).c_str());
+            assert_that(msg).is_rows().with_rows(get_result_rows(start_index, end_index));
+            
+            // tokens smaller than 0
+            msg = cquery_nofail(e, "SELECT * FROM t WHERE v = 1 AND token(pk) < 0 ALLOW FILTERING");
+            assert_that(msg).is_rows().with_rows(get_result_rows(0, 4));
+
+            // tokens larger than 0
+            msg = cquery_nofail(e, "SELECT * FROM t WHERE v = 1 AND token(pk) > 0 ALLOW FILTERING");
+            assert_that(msg).is_rows().with_rows(get_result_rows(5, testset_pks.size() - 1));
+
+            // token equality
+            msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) = {} ALLOW FILTERING", testset_tokens[3]).c_str());
+            assert_that(msg).is_rows().with_rows(get_result_rows(3, 3));
+
+            // empty range
+            msg = cquery_nofail(e, "SELECT * FROM t WHERE v = 1 AND token(pk) < 5 AND token(pk) > 100 ALLOW FILTERING");
+            assert_that(msg).is_rows().with_size(0);
+        };
+
+        eventually(do_tests);
+
+        cquery_nofail(e, "CREATE INDEX t_global ON t(v)");
+
+        eventually(do_tests);
+
+        cquery_nofail(e, "DROP INDEX t_global");
+        cquery_nofail(e, "CREATE INDEX t_local ON t((pk), v)");
```

Comment:
Would it also make sense to test with multiple indices or partition-column indices?

Reply:
I don't think it would make sense to create index on partition column, as no queries could then use this index - cannot mix `TOKEN` restrictions and non-token restrictions (which would be required to cause indexed select).

In the amended commit, I added a test with multiple (2) indexes.

Reply:
(I am speaking of `test_select_with_token_range_filtering`; `test_select_with_token_range_cases` is intended just for validating how different types of ranges are supported).

Reply:
OK

Reply:
@avelanarius I think you can create an index on one of *several* parition key columns (when the partition key is compound).

Reply:
@nyh Yes, but if I added a test with index on one of partition key columns, I still wouldn't be able to run a query that has both `TOKEN` restriction and restriction on indexed partition column (you cannot mix those restrictions). And this is exactly the type of query I'm running in this test (it wouldn't make sense to just test `TOKEN` restriction without index or just index query without `TOKEN`).